test/integration/extension_discovery_integration_test.cc

```diff
@@ -731,5 +731,59 @@ TEST_P(ExtensionDiscoveryIntegrationTest, BasicFailTerminalFilterNotAtEndOfFilte
   EXPECT_EQ("500", response->headers().getStatusValue());
 }
 
+// Validate that deleting listeners does not break active ECDS subscription.
+TEST_P(ExtensionDiscoveryIntegrationTest, ReloadBoth) {
```

Comment:
I understand that this is not part of this PR, but I think that the current integration test doesn't exercise the `onConfigRemoved` path because the test doesn't use delta-xDS.

Reply:
Ack, let's do this separately since this looks like some existing debt.